x-pack/plugins/ml/public/application/components/ml_inference/components/on_failure_configuration.tsx

```diff
@@ -0,0 +1,220 @@
+
+import {
+  EuiButtonEmpty,
+  EuiCode,
+  EuiCodeBlock,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFormRow,
+  EuiLink,
+  EuiSpacer,
+  EuiSwitch,
+  EuiSwitchEvent,
+  EuiText,
+  EuiTitle,
+} from '@elastic/eui';
+import { CodeEditor } from '@kbn/kibana-react-plugin/public';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
+
+import { SaveChangesButton } from './save_changes_button';
+import type { MlInferenceState } from '../types';
+import { getDefaultOnFailureConfiguration } from '../state';
+import { cancelEditMessage, editMessage } from '../constants';
+import { useMlKibana } from '../../../contexts/kibana';
+import { isValidJson } from '../../../../../common/util/validation_utils';
+
+interface Props {
+  handleAdvancedConfigUpdate: (configUpdate: Partial<MlInferenceState>) => void;
+  ignoreFailure: boolean;
+  onFailure: MlInferenceState['onFailure'];
+}
+
+export const OnFailureConfiguration: FC<Props> = ({
+  handleAdvancedConfigUpdate,
+  ignoreFailure,
+  onFailure,
+}) => {
+  const {
+    services: {
+      docLinks: { links },
+    },
+  } = useMlKibana();
+
+  const [editOnFailure, setEditOnFailure] = useState<boolean>(false);
+  const [isOnFailureValid, setIsOnFailureValid] = useState<boolean>(false);
+  const [onFailureString, setOnFailureString] = useState<string>(
+    JSON.stringify(onFailure, null, 2)
+  );
+
+  const updateIgnoreFailure = (e: EuiSwitchEvent) => {
+    handleAdvancedConfigUpdate({ ignoreFailure: e.target.checked });
+  };
+
+  const updateOnFailure = () => {
+    handleAdvancedConfigUpdate({ onFailure: JSON.parse(onFailureString) });
+    setEditOnFailure(false);
+  };
+
+  const handleOnFailureChange = (json: string) => {
+    setOnFailureString(json);
+    const valid = isValidJson(json);
+    setIsOnFailureValid(valid);
+  };
+
+  const resetOnFailure = () => {
+    setOnFailureString(JSON.stringify(getDefaultOnFailureConfiguration(), null, 2));
+    setIsOnFailureValid(true);
+  };
+
+  return (
+    <EuiFlexGroup direction="column">
+      {/* IGNORE FAILURE */}
+      <EuiFlexItem>
+        <EuiFlexGroup>
+          <EuiFlexItem grow={3}>
+            <EuiTitle size="s">
+              <h4>
+                {i18n.translate(
+                  'xpack.ml.trainedModels.content.indices.pipelines.addInferencePipelineModal.steps.advanced.onFailureTitle',
+                  { defaultMessage: 'Ingesting problematic documents' }
+                )}
+              </h4>
+            </EuiTitle>
+            <EuiSpacer size="m" />
+            <EuiText color="subdued" size="s">
+              <p>
+                <FormattedMessage
+                  id="xpack.ml.trainedModels.content.indices.pipelines.addInferencePipelineModal.steps.advanced.handleFailuresExplanation"
+                  defaultMessage="If the model fails to produce a prediction, e.g., due to the data schema change, the document will be ingested without the prediction."
```

Comment:
```suggestion
                  defaultMessage="If the model fails to produce a prediction, for example, due to the data schema change, the document will be ingested without the prediction."
```

Reply:
Updated in b0aa5e0d8ecc48b474dcd709982536fe852dcd9a